meetings/working-groups/collection-literals/Supporting-Interface-TargetTypes.md

```diff
@@ -0,0 +1,72 @@
+
+   Note that natural types do also appear both for dynamic and object.  However, It does not feel like these cases are substantively different or interesting enough to warrant a different 'natural type' (though the point is open for debate).
+
+   Conversely, we expect interface-target types to be most commonly used when passing data to other APIs, or when used to store data for properties the user has (and exposes) of those interface types.  In these cases, mutability is undesirable.  And, indeed, can be a source of problems (as external consumers might be able to change your data).
+
+   Because they core scenarios differ so wildly on their expectations around mutability and around the problems that arise (for either) by aligning on what the other does, it seems reasonable to state and explain things if we go down a path where there are different types and choices between natural and target types.
+
+4. Furthermore, alignment between the two systems (target and natural typing) could actually be highly negative if we promise a concrete collection type. For example, if we promise a `List<T>` type, then that eliminates the ability to use a singleton for the *very* common and *core* use-case of passing `[]` as an argument to `IEnumerable<T>` apis (or initializing an `IEnumerable<T>` property).  This would force users away from this, and require them to be as verbose as today to be explicit to get the desired savings.  This deeply undercuts are core narrative that you can use collection expressions and they can be the *efficient* type you can depend on.
+
+   Conversely, by not promising a concrete type, we get more opportunities to optimize, even for *non* empty cases.  For example, with:
+
+      ```c#
+      Goo(IEnumerable<int> values);
+      Goo([1, 2, 3])
+      ```
+
+   We can optimize in a way similar to what we do with `yield`ing iterators.  Namely, you would an `IEnumerable<T>` instance that was itself its own `IEnumerator<T>` instance (in the case where it has not been enumerated yet, and was being enumerated on the same thread that created it).  This would again be much more efficient than what you get today with any normal concrete type a user uses here (lists, arrays, etc.)   This again ties heavily into our story that by using collection expressions you can trust you're going to get the best results.
```

Comment:
```suggestion
   We can optimize in a way similar to what we do with `yield`ing iterators.  Namely, you would an `IEnumerable<T>` instance that was itself its own `IEnumerator<T>` instance (in the case where it has not been enumerated yet, and was being enumerated on the same thread that created it).  This would again be much more efficient than what you get today with any normal concrete type a user uses here (lists, arrays, etc.), and again ties heavily into our story that by using collection expressions you can trust you're going to get the best results.
```